### PR TITLE
feat(390): Add JSONPlaceholder configurations

### PR DESCRIPTION
**Pull request by onyedikachi-david**
**Summary:**  
Add JSONPlaceholder example configurations for GraphQL, YAML, and JSON formats.

**Issue Reference(s):**  
Related to tailcallhq/tailcallhq.github.io#390

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [x] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

**Additional Context:**
This PR adds example configurations for the JSONPlaceholder API as discussed in tailcallhq/tailcallhq.github.io#390. These examples demonstrate how to set up a Tailcall configuration for a complex API with multiple interconnected resources.

The configurations are added in three formats:
- GraphQL: `examples/jsonplaceholder.graphql`
- YAML: `examples/jsonplaceholder.yml`
- JSON: `examples/jsonplaceholder.json`

These examples will be referenced in the documentation (tailcallhq/tailcallhq.github.io) to provide users with a comprehensive starting point for using Tailcall with a real-world API.

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

**Comment by tusharmath** [comment]
Getting the following error message:

```
 tailcall start ./examples/jsonplaceholder.graphql
 INFO Env file: "/Users/tushar/Documents/Projects/tailcall/.env" loaded
 WARN A new release of tailcall is available: v0.104.0 ➜ v0.104.18
 WARN To upgrade, run: npm update -g @tailcallhq/tailcall
 WARN https://github.com/tailcallhq/tailcall/releases/tag/v0.104.18
 INFO File read: ./examples/jsonplaceholder.graphql ... ok
 INFO N + 1 detected: 34
ERROR Invalid Configuration
Caused by:
  • argument 'album' is a nullable type [at Album.photos.@call.@http.path]
  • argument 'user' is a nullable type [at ComplexQuery.userWithPostsAndComments.@call.@http.path]
  • argument 'post' is a nullable type [at ComplexQuery.userWithPostsAndComments.@call.@http.path]
  • argument 'post' is a nullable type [at Post.comments.@call.@http.path]
  • argument 'user' is a nullable type [at Query.albums.@http.path]
  • argument 'post' is a nullable type [at Query.comments.@http.path]
  • argument 'album' is a nullable type [at Query.photos.@http.path]
  • argument 'user' is a nullable type [at Query.posts.@http.path]
  • argument 'user' is a nullable type [at Query.todos.@http.path]
  • argument 'user' is a nullable type [at User.albums.@call.@http.path]
  • argument 'user' is a nullable type [at User.posts.@call.@http.path]
  • argument 'user' is a nullable type [at User.todos.@call.@http.path]
  ```

**Comment by tusharmath** [comment]
Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

**Comment by onyedikachi-david** [comment]
> Thanks @onyedikachi-david ! I tried running the configuration as above and got these errors:
> 
> ```
>  tailcall start ./examples/jsonplaceholder.graphql
>  INFO Env file: "/Users/tushar/Documents/Projects/tailcall/.env" loaded
>  WARN A new release of tailcall is available: v0.104.0 ➜ v0.104.18
>  WARN To upgrade, run: npm update -g @tailcallhq/tailcall
>  WARN https://github.com/tailcallhq/tailcall/releases/tag/v0.104.18
>  INFO File read: ./examples/jsonplaceholder.graphql ... ok
>  INFO N + 1 detected: 34
> ERROR Invalid Configuration
> Caused by:
>   • argument 'album' is a nullable type [at Album.photos.@call.@http.path]
>   • argument 'user' is a nullable type [at ComplexQuery.userWithPostsAndComments.@call.@http.path]
>   • argument 'post' is a nullable type [at ComplexQuery.userWithPostsAndComments.@call.@http.path]
>   • argument 'post' is a nullable type [at Post.comments.@call.@http.path]
>   • argument 'user' is a nullable type [at Query.albums.@http.path]
>   • argument 'post' is a nullable type [at Query.comments.@http.path]
>   • argument 'album' is a nullable type [at Query.photos.@http.path]
>   • argument 'user' is a nullable type [at Query.posts.@http.path]
>   • argument 'user' is a nullable type [at Query.todos.@http.path]
>   • argument 'user' is a nullable type [at User.albums.@call.@http.path]
>   • argument 'user' is a nullable type [at User.posts.@call.@http.path]
>   • argument 'user' is a nullable type [at User.todos.@call.@http.path]
> ```

Thanks for your review sir. Just fixed it.

File: examples/jsonplaceholder.graphql [hunk 1]
```diff
@@ -1,28 +1,281 @@
...
 }
 
+# Define the Query type with various fields
 type Query {
-  posts: [Post] @http(path: "/posts")
-  users: [User] @http(path: "/users")
+  # Fetch a post by ID
+  post(id: Int!): Post @http(path: "/posts/{{.args.id}}")
+  # Fetch all posts by a specific user
+  posts(user: Int!): [Post] @http(path: "/posts?userId={{.args.user}}")
```
Comment (tusharmath) [inline examples/jsonplaceholder.graphql:16]:
```suggestion
  posts(userId: Int): [Post] @http(path: "/posts?userId={{.args.user}}")
```
User should be optional. This is true for all the types that return a list of values. Please update the arguments and drop required.

File: examples/jsonplaceholder.graphql [hunk 2]
```diff
@@ -1,28 +1,281 @@
...
+  # Fetch a comment by ID
+  comment(id: Int!): Comment @http(path: "/comments/{{.args.id}}")
+  # Fetch all comments on a specific post
+  comments(post: Int!): [Comment] @http(path: "/comments?postId={{.args.post}}")
```
Comment (tusharmath) [inline examples/jsonplaceholder.graphql:20]:
```suggestion
  comments(postId: Int, email: String): [Comment] @http(path: "/comments?postId={{.args.post}}")
```

File: examples/jsonplaceholder.graphql [hunk 3]
```diff
@@ -1,28 +1,281 @@
...
+  # Fetch a user by ID
   user(id: Int!): User @http(path: "/users/{{.args.id}}")
+  # Fetch all users
+  users: [User] @http(path: "/users")
+}
+
+# Define the Mutation type with various fields for creating, updating, and deleting resources
+type Mutation {
+  # Create a new post
+  createPost(input: PostInput!): Post @http(method: "POST", path: "/posts", body: "{{.args.input}}")
```
Comment (tusharmath) [inline examples/jsonplaceholder.graphql:42]:
```suggestion
  createPost(input: PostInput!): Post @http(method: POST, path: "/posts", body: "{{.args.input}}")
```

Methods are not set in string. They are enums.

**Comment by tusharmath** [comment]
Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

**Pull request closed without merging**